pennylane/labs/tests/resource_estimation/test_resources_base.py

```diff
@@ -0,0 +1,313 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains tests for the Resources container class.
+"""
+
+from collections import defaultdict
+
+import pytest
+
+from pennylane.labs.resource_estimation.qubit_manager import QubitManager
+from pennylane.labs.resource_estimation.resources_base import (
+    Resources,
+    _combine_dict,
+    _scale_dict,
+    add_in_parallel,
+    add_in_series,
+    mul_in_parallel,
+    mul_in_series,
+)
+
+# pylint: disable= no-self-use,too-few-public-methods,comparison-with-itself
+
+
+class DummyResOp:
+    """A dummy class to populate the gate types dictionary for testing."""
+
+    def __init__(self, name):
+        """Initialize dummy class."""
+        self.name = name
+
+    def __hash__(self):
+        """Custom hash which only depends on instance name."""
+        return hash(self.name)
```

Comment:
```suggestion
from dataclasses import dataclass

@dataclass(frozen=True)
class DummyResOp:
    """A dummy class to populate the gate types dictionary for testing."""
    name: str
```

Something like this might be cleaner.